ffi/ffi_serialization.tcc

```diff
@@ -0,0 +1,195 @@
+#ifndef __LIBFF_FFI_FFI_SERIALIZATION_TCC__
+#define __LIBFF_FFI_FFI_SERIALIZATION_TCC__
+
+#include <libff/algebra/fields/fp.hpp>
+#include "ffi_serialization.hpp"
+
+namespace libff
+{
+
+namespace ffi
+{
+
+namespace internal
+{
+
+// Generic big-endian field_serializer for multi-component field
+// elements.
+template<typename FieldT>
+class field_serializer
+{
+public:
+    static bool read_from_buffer(
+        FieldT &f, const void *buffer, size_t buffer_size)
+    {
+        constexpr size_t component_size = sizeof(f.coeffs[0]);
+        size_t i = FieldT::extension_degree() - 1;
+        do {
+            if (buffer_size < component_size ||
+                !field_element_read(f.coeffs[i], buffer, component_size)) {
```

Comment:
See earlier comment. I'd probably break after the closing `)` to clearly separate condition and "body" in the if `block` (also applicable below)

Reply:
added line break